### PR TITLE
feat: Search panel

### PR DESCRIPTION
**Pull request by diaozhongpu**
only goto hospital page

### CODE REVIEW

**Comment by diaozhongpu** [comment]
hotfix for line111, 125@app/src/main/java/cn/hospital/registerplatform/data/dto/HospitalInfo.kt from #16

**Comment by diaozhongpu** [comment]
image
image
image
image

File: app/src/main/java/cn/hospital/registerplatform/data/repository/HospitalRepository.kt [hunk 1]
```diff
@@ -18,11 +18,12 @@ class HospitalRepository(
         hospitalApi.getHospitalList(loadType, page, size)
     }
 
-    fun getHospitalFilterList(type: String) = getRawResultList(pagingConfig) { loadType, page, size ->
+    fun getHospitalFilterList(name: String, type: String) = getRawResultList(pagingConfig) { loadType, page, size ->
         if (type == "所有") {
-            hospitalApi.getHospitalList(loadType, page, size)
+            // hospitalApi.getHospitalList(loadType, page, size)
+            hospitalApi.getHospitalFilterList(Gson().toJson(mapOf("name" to name)), loadType, page, size)
```
Comment (Borealin) [inline app/src/main/java/cn/hospital/registerplatform/data/repository/HospitalRepository.kt:24]: No need to use gson

File: app/src/main/java/cn/hospital/registerplatform/ui/component/hospital/HospitalListActivity.kt [hunk 2]
```diff
@@ -71,13 +78,18 @@ class HospitalListActivity : ActionBarActivity("医院列表") {
...
-                    getList(conditionArray[position])
+                    searchPosition = position
+                    getList(searchName, conditionArray[position])
 //                    ToastUtils.show(this@HospitalListActivity, conditionArray[position])
                 }
 
                 override fun onNothingSelected(parent: AdapterView<*>?) {
                 }
             }
+            hospitalSearchName.afterTextChanged { text ->
```
Comment (Borealin) [inline app/src/main/java/cn/hospital/registerplatform/ui/component/hospital/HospitalListActivity.kt:89]: could use data binding's bidirectional binding

File: app/src/main/java/cn/hospital/registerplatform/ui/component/main/HomeFragment.kt [hunk 3]
```diff
@@ -22,8 +25,14 @@ class HomeFragment : Fragment(R.layout.fragment_home) {
...
             }
-            searchContainer.setOnClickListener {
-                ToastUtils.show(requireContext(), R.string.waiting_for_backend_complete)
+            searchContainer.apply {
+                // startActivity(HospitalListActivity.newIntent(MainActivity))
+//                it.context.apply {
+//                    startActivity(HospitalListActivity.newIntent(this))
+//                }
+                setOnClickListener {
+                    startActivity(HospitalListActivity.newIntent(view.context))
```
Comment (Borealin) [inline app/src/main/java/cn/hospital/registerplatform/ui/component/main/HomeFragment.kt:34]: maybe use requireContext instead of view.context

**Review by Borealin — CHANGES_REQUESTED** [review]

**Review by Borealin — APPROVED** [review]

**Pull request merged**